apps/rehabstod/src/components/Form/NumberInput.tsx

```diff
@@ -1,49 +1,36 @@
-import { IDSInput } from '@frontend/ids-react-ts'
-import { ChangeEvent } from 'react'
+import { ChangeEvent, useId } from 'react'
+import { classNames } from '../../utils/classNames'
 import { TooltipIcon } from '../TooltipIcon/TooltipIcon'
+import { Input } from './Input'
 
 export function NumberInput({
   label,
   onChange,
   onBlur,
   description,
-  id,
   value,
-  isRange,
   max,
   min,
-  className,
-  classNameInput,
-  novalidation,
+  error,
+  inline = false,
 }: {
   label: string
   onChange: (event: ChangeEvent<HTMLInputElement>) => void
   onBlur?: () => void
   description?: string
-  id: string
   value: number | string
-  isRange?: boolean
+  inline?: boolean
   max?: number | string
   min?: number | string
-  className?: string
-  classNameInput?: string
```

Comment:
`className` on reusable components is problematic when refactoring later. It's difficult to know how a component is used and looks in different places when it's allowed to be changed with styling.

Some guideline
- Make sure a component always uses the full width and wrap with an element that constrains the component to fit within the layout.
- Add flags that describes the different states of a component, like with `error` here.
- Add types or styles to describe different styles, look at `DynamicLink` for example.